bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/util/JsonPatchHelper.java

```diff
@@ -0,0 +1,152 @@
+/********************************************************************************
+ * Copyright (c) 2022 STMicroelectronics..
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.emfcloud.modelserver.emf.util;
+
+import org.apache.log4j.Logger;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.change.ChangeDescription;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.emf.transaction.RollbackException;
+import org.eclipse.emf.transaction.Transaction;
+import org.eclipse.emf.transaction.TransactionalEditingDomain;
+import org.eclipse.emf.transaction.impl.InternalTransaction;
+import org.eclipse.emfcloud.modelserver.command.CCommandExecutionResult;
+import org.eclipse.emfcloud.modelserver.common.codecs.Codec;
+import org.eclipse.emfcloud.modelserver.common.codecs.EncodingException;
+import org.eclipse.emfcloud.modelserver.common.patch.AbstractJsonPatchHelper;
+import org.eclipse.emfcloud.modelserver.emf.common.ModelResourceManager;
+import org.eclipse.emfcloud.modelserver.emf.common.ModelServerEditingDomain;
+import org.eclipse.emfcloud.modelserver.emf.common.codecs.JsonCodecV2;
+import org.eclipse.emfcloud.modelserver.emf.configuration.ServerConfiguration;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.fge.jsonpatch.diff.JsonDiff;
+
+/**
+ * A Helper to create EMF Commands from a Json Patch.
+ */
+public class JsonPatchHelper extends AbstractJsonPatchHelper {
+
+   protected static final Logger LOG = Logger.getLogger(JsonPatchHelper.class.getSimpleName());
+
+   private final ModelResourceManager modelManager;
+   private final ServerConfiguration serverConfiguration;
+
+   public JsonPatchHelper(final ModelResourceManager modelManager, final ServerConfiguration serverConfiguration) {
+      this.modelManager = modelManager;
+      this.serverConfiguration = serverConfiguration;
+   }
+
+   @Override
+   protected TransactionalEditingDomain getEditingDomain(final EObject eObject) {
+      ResourceSet resourceSet = eObject.eResource().getResourceSet();
+      return modelManager.getEditingDomain(resourceSet);
+   }
+
+   @Override
+   protected Resource getResource(final String modelURI, final URI resourceURI) {
+      // FIXME We should make sure that modelURI and resourceURI are part of the same
```

Comment:
create an issue?

Reply:
This should probably be handled with #159